### PR TITLE
Fix: Gemstones item query

### PR DESCRIPTION
**Pull request by Thunderblade73**
## What
Fixes the item query for gemstones. Especially for queued gfs.
I don't like the fix, but it works.

## Changelog Fixes
+ Queued gfs now works with gemstone names. - Thunderblade73

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/utils/StringUtils.kt [hunk 1]
```diff
@@ -20,11 +20,13 @@ object StringUtils {
     private val resetPattern = "(?i)§R".toPattern()
     private val sFormattingPattern = "(?i)§S".toPattern()
     private val stringColourPattern = "§[0123456789abcdef].*".toPattern()
+    private val asciiPattern = "[^${Char(0)}-${Char(255)}]".toPattern()
```
Comment (ThatGravyBoat) [inline src/main/java/at/hannibal2/skyhanni/utils/StringUtils.kt]: This regex is not for ASCII, ASCII does not go to 255 this regex should be changed to the following. using Char(n) does not provide any benefit here.

```suggestion
private val asciiPattern = "[^\\x00-\\x7F]".toPattern()
```

**Review by ThatGravyBoat — CHANGES_REQUESTED** [review]

File: src/main/java/at/hannibal2/skyhanni/utils/StringUtils.kt (hunk 1 above)
Comment (Thunderblade73) [reply src/main/java/at/hannibal2/skyhanni/utils/StringUtils.kt]: Yeah, made an ups by including 128 characters more 😅. Also I didn't know how to but a specific char into a String in Kotlin, so thank you.

File: src/main/java/at/hannibal2/skyhanni/utils/StringUtils.kt [hunk 2]
```diff
@@ -20,11 +20,13 @@ object StringUtils {
     private val resetPattern = "(?i)§R".toPattern()
     private val sFormattingPattern = "(?i)§S".toPattern()
     private val stringColourPattern = "§[0123456789abcdef].*".toPattern()
+private val asciiPattern = "[^\\x00-\\x7F]".toPattern()
```
Comment (jani270) [inline src/main/java/at/hannibal2/skyhanni/utils/StringUtils.kt]:
```suggestion
    private val asciiPattern = "[^\\x00-\\x7F]".toPattern()
```

**Review by jani270 — CHANGES_REQUESTED** [review]
🤓

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**